src/GraphQL.ApiTests/GraphQL.approved.txt

```diff
@@ -1918,8 +1917,7 @@ namespace GraphQL.Types
     }
     public interface IProvideResolvedType
     {
-        GraphQL.Types.IGraphType ResolvedType { get; }
-        System.Type Type { get; }
+        GraphQL.Types.IGraphType ResolvedType { get; set; }
```

Comment:
I don't like the fact that the field is writable.
```suggestion
        GraphQL.Types.IGraphType ResolvedType { get; internal set; }
```

Reply:
Our goal should be to not let runtime code have write-access to the schema.  I understand we are a long way from that now, but I do not think we need this writable at this time.  Even the visitor you added _could_ have been written without it.

Reply:
![изображение](https://user-images.githubusercontent.com/21261007/111008031-483f1880-83a1-11eb-882a-7e016ded798a.png)

Breaking change for all implementations.

Reply:
Oh I forgot about that.  There's a feature request for that, since it's allowed by the .NET runtime, but not C# -- I can't find the link now.

Of course we could add explicit implementations for all classes that use `ResolvedType`, and then it would not be a breaking change -- either now or to make it settable in the future.

But here's the thing: do we really want to make this settable???

Reply:
> > We also are modifying the schema after it's been initialized and verified. There's no protection against modifying an initialized schema.
> 
> It's by design. The schema allows modification at any time. Nothing prevents, for example, change the descriptions of any field on the go. It's just for the responsibility of the caller. Changing the schema after initialization is an advanced scenario and I believe that the caller must understand the consequences. It is worth noting this in the documentation.

Ok